File: tests/fixtures/vulnerable/XFileHandler.java
// FIXTURE (cross-file): the source lives here.
import java.sql.*;
import javax.servlet.http.HttpServletRequest;

public class XFileHandler {
    public ResultSet handle(HttpServletRequest request, Statement stmt) throws SQLException {
        return XFileDb.runQueryXF(stmt, "SELECT * FROM users WHERE id = " + request.getParameter("id"));
    }
}

File: tests/fixtures/vulnerable/xfile_handler.go
// FIXTURE (cross-file): the source lives here.
package fixtures

import (
	"database/sql"
	"net/http"
)

func handlerXF(r *http.Request, db *sql.DB) {
	runQueryXF(db, "SELECT * FROM users WHERE id = "+r.FormValue("id"))
}
